zebra/zebra_rib.c

```diff
@@ -2745,6 +2748,24 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		if (CHECK_FLAG(same->status, ROUTE_ENTRY_REMOVED))
 			continue;
 
+		/* If kernel_reconcile is set, then mark Zebra-originated route
+		 * for deletion. Zebra learned these routes from kernel while
+		 * start up. Now, since we learned new route for same destination,
+		 * this is the time to clear stale route, so that both kernel
+		 * and FPM are updated while rib_process. This will result in no
+		 * traffic loss, if same route->NH is learned after startup.
+		 */
+		if (kernel_reconcile
+			&& CHECK_FLAG(same->flags, ZEBRA_FLAG_KERNEL_RECONCILE)) {
+			++zebra_kernel_reconcile_route_deleted;
+			if (IS_ZEBRA_DEBUG_RIB)
+				rnode_debug(rn, same->vrf_id,
+					"kernel_reconcile: match, A %lu D %lu rn %p",
```

Comment:
Having add/del key words instead of 'A' & 'D' would be better IMO.

Reply:
Sure, I will change it. Thanks.